src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -1264,6 +1264,11 @@ private void updateCurrentConnection(
   }
 
   private void updateHostIndex(List<HostInfo> latestTopology) throws SQLException {
+    if (this.currentHostIndex == NO_CONNECTION_INDEX) {
```

Comment:
HostInfo currentHost = this.currentHostIndex == NO_CONNECTION_INDEX ? null : this.hosts.get(this.currentHostIndex);

?